server/src/main/java/org/opensearch/cluster/routing/RoutingPool.java

```diff
@@ -58,6 +59,8 @@ public static RoutingPool getShardPool(ShardRouting shard, RoutingAllocation all
      * @return {@link RoutingPool} for the given index.
      */
     public static RoutingPool getIndexPool(IndexMetadata indexMetadata) {
-        return indexMetadata.isRemoteSnapshot() ? REMOTE_CAPABLE : LOCAL_ONLY;
+        return indexMetadata.isRemoteSnapshot()
+            || IndexModule.DataLocalityType.PARTIAL.name()
+                .equals(indexMetadata.getSettings().get(IndexModule.INDEX_STORE_LOCALITY_SETTING.getKey())) ? REMOTE_CAPABLE : LOCAL_ONLY;
```

Comment:
Can we add test for this change ?

Reply:
will do

Reply:
Should this be feature flagged?

Reply:
will add feature flag and move this change to https://github.com/neetikasinghal/OpenSearch/pull/144/files